### PR TITLE
feat: referendum vote screen

### PR DESCRIPTION
**Pull request by johnthecat**
Pop up_2

### CODE REVIEW

File: src/renderer/entities/governance/lib/votingService.ts [hunk 1]
```diff
@@ -135,19 +182,21 @@ const calculateAccountVotesTotalBalance = (votes: AccountVote[]) => {
...
 // Voted types
 
-const isStandardVote = (vote: AccountVote): vote is StandardVote => vote.type === 'standard';
+const isStandardVote = (vote: AccountVote): vote is StandardVote => vote.type === 'Standard';
 
-const isSplitVote = (vote: AccountVote): vote is SplitVote => vote.type === 'split';
+const isSplitVote = (vote: AccountVote): vote is SplitVote => vote.type === 'Split';
 
-const isSplitAbstainVote = (vote: AccountVote): vote is SplitAbstainVote => vote.type === 'splitAbstain';
+const isSplitAbstainVote = (vote: AccountVote): vote is SplitAbstainVote => vote.type === 'SplitAbstain';
```
Comment (sokolova-an) [inline src/renderer/entities/governance/lib/votingService.ts:199]: Why are all types now capitalised?

File: src/renderer/entities/governance/model/vote.ts [hunk 2]
```diff
@@ -0,0 +1 @@
+export const voteModel = {};
```
Comment (sokolova-an) [inline src/renderer/entities/governance/model/vote.ts:1]: do we need it or is it for the future?

File: src/renderer/features/governance/aggregates/details.ts [hunk 3]
```diff
@@ -12,6 +13,8 @@ import { votingAggregate } from './voting';
 
 const flow = createGate<{ chain: Chain; referendum: Referendum }>();
 
+const $canVote = walletModel.$activeWallet.map((wallet) => !walletUtils.isWatchOnly(wallet));
```
Comment (Asmadek) [inline src/renderer/features/governance/aggregates/details.ts]: we have permissionUtils for such checking

File: src/renderer/shared/core/types/transaction.ts [hunk 4]
```diff
@@ -44,7 +44,9 @@ export const enum TransactionType {
   REMOVE_PURE_PROXY = 'kill_pure_proxy',
 
   UNLOCK = 'unlock',
-  REMOVE_VOTE = 'remove_vote',
+  VOTE = 'vote',
+  REVOTE = 'revote',
+  RETRACT_VOTE = 'remove_vote',
```
Comment (sokolova-an) [inline src/renderer/shared/core/types/transaction.ts:49]: why rename it, the extrinsic call has name 'remove_vote'

File: src/renderer/features/governance/lib/createFeeCalculator.ts [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+      dropped: $dropped,
+    }),
+  });
+
+  sample({
+    clock: fetchFeeFx.doneData,
+    target: $fee,
+  });
+
+  return { $: $fee, $pending: fetchFeeFx.pending, drop: dropFee };
```
Comment (Asmadek) [inline src/renderer/features/governance/lib/createFeeCalculator.ts:81]: Maybe use our style guides, `{ $value, events: { event } }`

File: src/renderer/widgets/VoteModal/ui/VoteForm.tsx [hunk 6]
```diff
@@ -0,0 +1,255 @@
...
+              {t('governance.vote.abstainConvictionWarningDescription')}
+            </FootnoteText>
+          </div>
+        </ConfirmModal>
+      )}
+    </>
+  );
+};
+
+// const FeeSection = () => {
```
Comment (Asmadek) [inline src/renderer/widgets/VoteModal/ui/VoteForm.tsx]: add todo here

**Review by Asmadek — APPROVED** [review]

File: src/renderer/features/governance/lib/createFeeCalculator.ts [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+    clock: fetchFee,
+    fn: () => false,
+    target: $dropped,
+  });
+
+  sample({
+    clock: fetchFee,
+    source: {
+      pending: fetchFeeFx.pending,
+      // dropped: $dropped,
```
Comment (sokolova-an) [inline src/renderer/features/governance/lib/createFeeCalculator.ts:57]: do we need it here?

File: src/renderer/features/governance/aggregates/details.ts (hunk 3 above)
Comment (johnthecat) [reply src/renderer/features/governance/aggregates/details.ts]: good point, tnx

File: src/renderer/features/governance/lib/createFeeCalculator.ts (hunk 5 above)
Comment (johnthecat) [reply src/renderer/features/governance/lib/createFeeCalculator.ts:81]: Added todo to discuss code style later

**Pull request merged**